### PR TITLE
CORE-500 Handle transparency for Forward Proxy

### PR DESCRIPTION
**Pull request by zdmytriv**

### CODE REVIEW

File: src/main/java/org/littleshoot/proxy/impl/TraceableProxyHttpRequestEncoder.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    if (headerVgsCustomerB3TraceId == null) {
+      // looks like all tracing header were injected by VGS so removing them
+      headers.remove(TraceUtils.HEADER_X_B3_TRACEID);
+      headers.remove(TraceUtils.HEADER_X_B3_SPANID);
+      headers.remove(TraceUtils.HEADER_X_B3_SAMPLED);
+      headers.remove(TraceUtils.HEADER_X_B3_PARENTSPANID);
+    } else {
+      // preserving "X-B3-TraceId" header
+      headers.set(TraceUtils.HEADER_X_B3_TRACEID, headerVgsCustomerB3TraceId);
```
Comment (mjallday) [inline src/main/java/org/littleshoot/proxy/impl/TraceableProxyHttpRequestEncoder.java]: if i'm reading this correctly it looks like we're saying that we do _not_ remove `HEADER_VGS_CUSTOMER_B3_*` headers. is this the desired behavior?

Comment (zdmytriv) [reply src/main/java/org/littleshoot/proxy/impl/TraceableProxyHttpRequestEncoder.java]: we do. 

I was going back and forth on how we will preserve customer headers. 

We do remove these in https://github.com/verygoodsecurity/LittleProxy/pull/86/files#diff-f079e3781194aaacdf2967b5b9b08ab8bb9ea201e877b1af55be58a040414616R17-R21

File: src/main/java/org/littleshoot/proxy/impl/TracingAwareHttpRequestEncoder.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+    this.ctx = ctx;
+  }
+
+  @Override
+  protected void sanitizeHeadersBeforeEncode(HttpRequest msg, boolean isAlwaysEmpty) {
+    HttpHeaders headers = msg.headers();
+
+    if (isForwardProxyTransparent()) {
+      headers.remove(TraceUtils.HEADER_X_B3_TRACEID);
```
Comment (nestorsokil) [inline src/main/java/org/littleshoot/proxy/impl/TracingAwareHttpRequestEncoder.java:26]: Let's add `VGS-Request-Id` here too. We then can remove that code in vault.

Comment (zdmytriv) [reply src/main/java/org/littleshoot/proxy/impl/TracingAwareHttpRequestEncoder.java:26]: added

**Review by nestorsokil — APPROVED** [review]

**Pull request merged**